portal/server/routes/contract/about/industryFields/index.js

```diff
@@ -7,21 +7,31 @@ const industryFields = (postedSubmissionDetails, industrySectors) => {
 
   let industrySectorObj;
 
-  if (industrySectorCode) {
+  if (!industrySectorCode) {
+    submissionDetails['industry-sector'] = {};
+  } else {
     industrySectorObj = getIndustrySectorById(industrySectors, industrySectorCode);
 
-    submissionDetails['industry-sector'] = {
-      code: industrySectorCode,
-      name: getIndustrySectorById(industrySectors, industrySectorCode).name,
-    };
-  } else {
-    submissionDetails['industry-sector'] = {};
+    if (industrySectorObj) {
+      submissionDetails['industry-sector'] = {
+        code: industrySectorCode,
+        name: industrySectorObj.name,
+      };
+    } else {
+      /**
+       * This is required because - if an industry sector is not found
+       * and we save 'industry-sector' as undefined,
+       * the validation thinks that it has a valid value.
+       * Therefore, if no industry sector is found, the object should be wiped.
+       */
+      submissionDetails['industry-sector'] = {};
+    }
```

Comment:
After GOV design update - not sure why exactly, but this was erroring and causing validation issues if no indusry is selected - the validation would think that "undefined" is a valid value. (DTFS2-7640)

Therefore i've updated this so that if no industry sector is found by the provided ID, an empty object with be returned.

Reply:
Interesting -- Where is this validation failing? It's odd we allow undefined... Is the answer to do this or is the answer to update our validation in this area (with another ticket)?

Reply:
I'd be tempted to consider refactoring this -- but again maybe beyond the scope of this so we can leave this complexity for now and pick it up another time

Reply:
For what its worth -- We're not deep cloning the postedSubmissionDetails -- So we are changing both, so assignment is pointless
```
const industryFields = (submissionDetails, industrySectors) => {
    const industrySectorCode = submissionDetails['industry-sector'];  
    const industrySectorObj = industrySectorCode && getIndustrySectorById(industrySectors, industrySectorCode);  

    if(!industrySectorObj) {  
        submissionDetails['industry-sector'] = {  
            code: industrySectorCode,  
            name: industrySectorObj.name,  
        };       
    } else {
        submissionDetails['industry-sector'] = {};  
    }

    return submissionDetails;  
}
```

Reply:
I agree, lots of things could be improved.

I cannot remember the exact details for this off the top of my head - can dive back into previous commits if you'd like, but without this change, if I remember correctly:

- The validation thought "Select value" was a valid value.
- With the above in play, `submissionDetails['industry-sector']` was being set to `undefined`, because no `industrySectorCode` is found from "Select value".

More details in this ticket: https://ukef-dtfs.atlassian.net/browse/DTFS2-7640

In other instances, `submissionDetails['industry-sector']` is set to an empty object. I think it's safest to do this instead of allowing `undefined`

Reply:
Will have a look into those suggestions @AlexBramhill , thanks.

I was trying more for a "quick fix" with the existing function but more than happy to improve it.

I don't think the first suggestion will work as the object is returned before `industry-class` logic happens

Reply:
Am happy for leaving this for now as it's not departing from existing patterns in this file

Reply:
Made some updates - resolving this comment in favour of a new fresh thread. https://github.com/UK-Export-Finance/dtfs2/pull/3684/files#r1856629286